### ISSUE TITLE
Selection Routine called With `nil` With Array of Choices

### ISSUE THREAD

**Issue by zeroSteiner** [dmendel/bindata#122]
I'm modeling a data block where there is an array of types that correspond to an array of values. It's organized such that the first field is the count, followed by all the types, followed by all the values. I'm using an array of choices that uses a selection routine that references the first (type) array to determine the type of each member in the second. When doing this and initializing the class with both arrays populated, I get an exception due to the selection routine being called with `nil`.

Example class:
```ruby
class MemberTypeInfo < BinData::Record
  endian                   :little
  default_parameter        member_count: 0
  array                    :member_types, :type => :uint8, :initial_length => :member_count
  array                    :member_values, initial_length: -> { member_types.length } do
    choice :member_value, :selection => lambda { selection_routine(index) } do
      uint8                  8
      uint16                 16
      uint32                 32
    end
  end

  private

  def selection_routine(index)
    puts 'index is nil' if index.nil?
    member_types[index]
  end
end
```

Does not work:
```
mti = MemberTypeInfo.new(member_types: [8, 16], member_values: [0xff, 0xffff])
```

Does work:
```
mti = MemberTypeInfo.new(member_types: [8, 16])
mti.member_values[0] = 0xff
mti.member_values[1] = 0xffff
```

Is it possible to define the model to allow the `member_values` to be defined when initializing the class?

**Comment by dmendel** [comment]
Fixed in 2.4.7.

Thanks for the report.

**Comment by zeroSteiner** [comment]
Thank you very much!
